Plugins/Tweaks/Tweaks/AddPrestigeclassCasterLevels.cpp

```diff
@@ -0,0 +1,137 @@
+#include "Tweaks/AddPrestigeclassCasterLevels.hpp"
+
+#include "Services/Hooks/Hooks.hpp"
+#include "Utils.hpp"
+
+#include "API/CAppManager.hpp"
+#include "API/CNWSCreature.hpp"
+#include "API/CNWSCreatureStats.hpp"
+#include "API/CNWSRules.hpp"
+#include "API/CServerExoAppInternal.hpp"
+#include "API/CServerExoApp.hpp"
+#include "API/CTwoDimArrays.hpp"
+#include "API/Functions.hpp"
+#include "API/Globals.hpp"
+
+namespace Tweaks
+{
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+
+uint8_t AddPrestigeclassCasterLevels::s_classCasterType[Constants::ClassType::MAX + 1];
+uint8_t AddPrestigeclassCasterLevels::s_divModClasses[Constants::ClassType::MAX + 1];
+uint8_t AddPrestigeclassCasterLevels::s_arcModClasses[Constants::ClassType::MAX + 1];
+bool AddPrestigeclassCasterLevels::s_bAdjustCasterLevel = false;
+
+AddPrestigeclassCasterLevels::AddPrestigeclassCasterLevels(Services::HooksProxy* hooker)
+{
+    hooker->RequestSharedHook<Functions::_ZN17CNWSCreatureStats13GetClassLevelEhi, uint8_t>(&CNWSCreatureStats__GetClassLevel);
+    hooker->RequestSharedHook<Functions::_ZN25CNWVirtualMachineCommands28ExecuteCommandGetCasterLevelEii, int32_t>(&CNWVirtualMachineCommands__ExecuteCommandGetCasterLevel);
+    hooker->RequestSharedHook<Functions::_ZN25CNWVirtualMachineCommands25ExecuteCommandResistSpellEii, int32_t>(&CNWVirtualMachineCommands__ExecuteCommandResistSpell);
+    hooker->RequestSharedHook<Functions::_ZN11CGameEffect10SetCreatorEj, void>(&CGameEffect__SetCreator);
+    hooker->RequestSharedHook<Functions::_ZN8CNWRules13LoadClassInfoEv, void>(&CNWRules__LoadClassInfo);
+
+    for (int i = 0; i <= NWNXLib::API::Constants::ClassType::MAX; i++)
```

Comment:
Also, a standard way to get the number of elements in an  array would be:
```
for (int i = 0; i < sizeof(s_classCasterType)/sizeof(s_classCasterType[0]); i++
```

this will work if you modify the size above, but forget to modify the loop.